cdap-app-fabric/src/main/java/io/cdap/cdap/master/environment/k8s/MasterEnvironmentMain.java

```diff
@@ -113,7 +123,8 @@ public static void doMain(String[] args) throws Exception {
 
         RemoteClientFactory remoteClientFactory = new RemoteClientFactory(
           masterEnv.getDiscoveryServiceClientSupplier().get(),
-          new DefaultInternalAuthenticator(new WorkerAuthenticationContext()));
+          getInternalAuthenticator(cConf));
```

Comment:
Instead of running the sidecar and primary containers on the same main class, maybe it would be cleaner and more maintainable to separate them into two main classes? This can be done in a future PR as well.

Reply:
Ack